one-ui/src/main/java/com/marklogic/hub/curation/controllers/JobsController.java

```diff
@@ -0,0 +1,110 @@
+import com.marklogic.hub.oneui.managers.SearchableManager;
+import com.marklogic.hub.oneui.models.HubConfigSession;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import java.util.Collections;
+import java.util.List;
+
+@Controller
+@RequestMapping(value="/api/jobs")
+public class JobsController extends SearchableManager {
+    @Autowired
+    private HubConfigSession hubConfig;
+
+
+    @RequestMapping(method = RequestMethod.GET)
+    @ResponseBody
+    public ResponseEntity<?> getJobs(@RequestParam(value = "flowName", required = false) String flowName) {
+        JobDocManager jobDocManager = getJobDocManager();
+        ArrayNode jobsArray;
+        if (flowName == null) {
+            FlowManager flowManager = new FlowManagerImpl(hubConfig);
+            List<String> flowNames = flowManager.getFlowNames();
+            jobsArray = new ObjectMapper().createArrayNode();
+            flowNames.forEach((name) -> {
+                    jobsArray.addAll((ArrayNode) getJobs(jobDocManager, null, name));
+            });
+        } else {
+            jobsArray = (ArrayNode) getJobs(jobDocManager, null, flowName);
+        }
+        return new ResponseEntity<>(flattenJobsJson(jobsArray), HttpStatus.OK);
+    }
+
+    @RequestMapping(value = "/{jobId}", method = RequestMethod.GET)
+    @ResponseBody
+    public ResponseEntity<?> getJob(@PathVariable String jobId) {
+        JsonNode jobsObj = getJobs(getJobDocManager(), jobId, null);
+        return new ResponseEntity<>(flattenJobsJson(jobsObj), HttpStatus.OK);
+    }
+
+    @RequestMapping(value = "latest/{flowName}", method = RequestMethod.GET, produces = {MediaType.APPLICATION_JSON_VALUE})
+    @ResponseBody
+    public JsonNode getLatestJobDocForFlow(@PathVariable String flowName) {
+        return flattenJobsJson(getJobDocManager().getLatestJobDocumentForFlow(flowName));
+    }
+
+    @RequestMapping(value = "latest", method = RequestMethod.GET, produces = {MediaType.APPLICATION_JSON_VALUE})
+    @ResponseBody
+    public JsonNode getLatestJobDocsPerFlow() {
+        return flattenJobsJson(getJobDocManager().getLatestJobDocumentForFlows(Collections.emptyList()));
+    }
+
+    private JsonNode getJobs(JobDocManager jobDocManager, String jobId, String flowName) {
+        JsonNode jobsJson = (ArrayNode) jobDocManager.getJobDocument(jobId, flowName);
+        if (jobsJson == null) {
+            throw new RuntimeException("Unable to get job document");
```

Comment:
I recommend adding the jobId and the flowName to this message - e.g. "Unable to find document with ID ${jobId} for flow ${flowName}".